icu4j/main/tests/core/src/com/ibm/icu/dev/test/util/XLocaleMatcherTest.java

```diff
@@ -264,75 +255,214 @@ private long timeLocaleMatcher(String title, ULocale desired, LocaleMatcher matc
+                        Integer prevIndex = uniqueTests.get(inputs);
+                        if (prevIndex == null) {
+                            uniqueTests.put(inputs, tests.size());
+                        } else {
+                            System.out.println("Locale matcher test case on line " + test.lineNr
+                                    + " is a duplicate of line " + tests.get(prevIndex).lineNr);
+                        }
+                        tests.add(test.clone());
+                    } else {
+                        throw new IllegalArgumentException("test data syntax error on line "
+                                + test.lineNr + "\n" + line);
+                    }
+                }
+            }
+        }
+        System.out.println("Number of duplicate locale matcher test cases: " + (tests.size() - uniqueTests.size()));
+        return tests;
+    }
 
-            String expectedUi = arguments.size() < 4 ? null : arguments.get(3);
-            String expectedUiLanguageTag = expectedUi == null || expectedUi.equals("null") ? null
-                : new ULocale(expectedUi).toLanguageTag();
+    private static ULocale getULocaleOrNull(String s) {
+        if (s.equals("null")) {
+            return null;
+        } else {
+            return new ULocale(s);
+        }
+    }
 
-            if (breakpoint) {
-                breakpoint = false; // put debugger breakpoint here to break at @debug in test file
+    @Test
+    @Parameters(method = "readTestCases")
+    public void dataDriven(TestCase test) {
+        XLocaleMatcher matcher;
+        if (test.def.isEmpty() && test.distance.isEmpty() && test.threshold.isEmpty()) {
+            matcher = new XLocaleMatcher(test.supported);
+        } else {
+            XLocaleMatcher.Builder builder = XLocaleMatcher.builder();
+            builder.setSupportedLocales(test.supported);
+            if (!test.def.isEmpty()) {
+                builder.setDefaultLanguage(new ULocale(test.def));
             }
-            XLocaleMatcher matcher = threshold < 0 && distanceOption == DistanceOption.REGION_FIRST
-                ? newXLocaleMatcher(supportedList)
-                : newXLocaleMatcher(supportedList, threshold, distanceOption);
-            commentBase = "(" + lineNumber + ") " + commentBase;
-
-            ULocale bestSupported;
-            if (expectedUi != null) {
-                bestSupported = matcher.getBestMatch(desiredList, bestDesired);
-                ULocale bestUI = XLocaleMatcher.combine(bestSupported, bestDesired.value);
-                assertEquals(commentBase + " (UI)", expectedUiLanguageTag, bestUI == null ? null : bestUI.toLanguageTag());
-            } else {
-                bestSupported = matcher.getBestMatch(desiredList);
+            if (!test.distance.isEmpty()) {
+                DistanceOption distance;
+                switch (test.distance) {
+                case "normal":
+                    distance = DistanceOption.REGION_FIRST;
```

Comment:
We should go ahead and and change REGION_FIRST to LANGUAGE_FIRST

Reply:
Noted, but I prefer to do that for a different ticket.